### PR TITLE
images: Add workaround for systemd 250

### LINKED ISSUE #2788: Image refresh for arch
Image refresh for arch
 * [ ] FAIL: image-refresh arch

### PR DESCRIPTION
**Pull request by jelly**
Since systemd 250 it seems sshd waits on network-online.target if
reflector-init.service exists which has a Before=sshd.service and
Wants/After=network-online.target. This makes sshd start after a 2
minute systemd-networkd-wait-online timeout. The issue has not yet been
reported upstream.

 * [x] image-refresh arch

Fixes #2788

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on 14-cockpit-aws-tasks.
Log: https://logs.cockpit-project.org/logs/image-refresh-2796-20220111-170808/

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20220111-171717

**Comment by martinpitt** [comment]
Still seems to have trouble booting, now for second boot? Although not all tests are slow, just a few.

**Comment by jelly** [comment]
> 

Interesting, it had trouble booting in the first place due to network-online.target being triggered by a systemd unit, so maybe that's the case in a test.

**Comment by jelly** [comment]
> > 
> 
> Interesting, it had trouble booting in the first place due to network-online.target being triggered by a systemd unit, so maybe that's the case in a test.

I have the image locally, will investigate.

**Comment by jelly** [comment]
Locally `TEST_OS=arch ./test/common/run-tests --test-dir test/verify TestTimers.testCreate -vt` works fine, TestJournal.testRebootAndTime fails reproduceable locally.

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-2796-20220119-070808/

**Comment by cockpituous** [comment]
Command '('git', 'push', 'origin', '+HEADNone/heads/image-refresh-arch-20220119-071919')' returned non-zero exit status 1.: https://logs.cockpit-project.org/logs/image-refresh-2796-20220119-070808/

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-2796-20220119-072020/

**Comment by cockpituous** [comment]
Command '('git', 'push', 'origin', '+HEADNone/heads/image-refresh-arch-20220119-073131')' returned non-zero exit status 1.: https://logs.cockpit-project.org/logs/image-refresh-2796-20220119-072020/

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-2796-20220119-082929/

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20220119-084040

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-2796-20220119-084141/

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20220119-085252

File: images/scripts/arch.setup [hunk 1]
```diff
@@ -112,5 +112,11 @@ pacman -Scc --noconfirm
 # Reduce image size, remove cloud-init, reflector
 pacman -Rscn --noconfirm cloud-init reflector
 
+# Remove the no longer required systemd units as they are not required and seem
+# to cause an issue delaying sshd to be started after network online target
+# timeout. As reflector-init is reverse dependency on sshd which needs network-online.target
+rm /etc/systemd/system/reflector-init.service
```
Comment (martinpitt) [inline images/scripts/arch.setup:118]: The command above removes the "reflector" package. How come that its service is still enabled? Doesn't package removal clean up after itself properly?

**Comment by martinpitt** [comment]
Still lots of [VM boot trouble](https://logs.cockpit-project.org/logs/pull-2796-20220119-093722-e7e976d9-arch-cockpit-project-cockpit/log.html#9)

File: images/scripts/arch.setup (hunk 1 above)
Comment (jelly) [reply images/scripts/arch.setup:118]: Ah! This is a custom added systemd unit for the cloud image, which triggered a systemd bug which I probably should report. Described in the description.

**Comment by jelly** [comment]
Maybe the issue is systemd-networkd being stuck in configuring:

```
Jan 18 12:27:53 archlinux systemd-networkd[250]: eth0: Received remembered DHCPv4 route (configured): dst: 172.27.0.3/32, src: n/a, gw: n/a, prefs>
Jan 18 12:27:53 archlinux systemd-networkd[250]: eth0: DHCPv4 address and routes set.
Jan 18 12:27:53 archlinux systemd-networkd[250]: eth0: link_check_ready(): IPv6LL is not configured yet.
```

So locally disabled IPv6LocalAddressGeneration.

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-9fgq4.
Log: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-2796-20220124-083333/

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20220124-084444

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-9fgq4.
Log: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-2796-20220124-084444/

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20220124-085656

**Comment by jelly** [comment]
All the retries indicate there is still an issue with rebooting the machien.

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20220124-180909

**Comment by marusak** [comment]
`Error: NetworkManager is not running.`

**Comment by jelly** [comment]
eported upstream systemd/systemd#22237

**Comment by martinpitt** [comment]
Indeed: Even with `no-auto-default=*`, NM grabs eth1:
```
# nmcli d
DEVICE  TYPE      STATE                   CONNECTION 
eth0    ethernet  connected (externally)  eth0       
eth1    ethernet  disconnected            --         
lo      loopback  unmanaged               --         

device (eth1): state change: unavailable -> disconnected (reason 'none', sys-iface-state: 'managed')
```
and it also seems to manage eth0 now, so networkd and NM will definitively collide. I found a different way of making NM ignore these.

**Pull request closed without merging**